openedx/openedx-platform
# Discussion fixes from solutions

**Description:** Solutions fork contains some fixes/improvements  for Discussions JS code - this PR aims to pull them into upstream edx-platform

**JIRA:** [YONK-328](https://openedx.atlassian.net/browse/YONK-328)

**Sandboxes:** [LMS](http://pr12914.sandbox.opencraft.hosting/), [Studio](http://studio-pr12914.sandbox.opencraft.hosting/)

**Testing instructions:**

Short: No changes in functionality are added - running test suite and smoke-testing everything as single user should be enough.

Detailed:
1. Create&publish a course
2. Log in to LMS, go to course
3. Go to "Discussions" tab
4. Create two-three new posts of different types (question/discussion)
5. Check navigation works - jump between threads. Refresh the page and make sure the same thread is displayed after refresh
6. (optionally, log in as a different user) Add some comments and responses.
7. Try deleting comments/responses/threads
8. Log in as different user - try upvoting threads
9. As thread/response/comment owner - try editing contents

Reviewers:
- [x] @robrap 
- [x] @bjacobel
